integration/framework/framework.go

```diff
@@ -169,40 +198,50 @@ type DockerRunArgs struct {
 	Args []string
 }
 
+// TODO(vmarmol): Refactor a set of "RunCommand" actions.
 // Runs a Docker container in the background. Uses the specified DockerRunArgs and command.
 //
 // e.g.:
 // RunDockerContainer(DockerRunArgs{Image: "busybox"}, "ping", "www.google.com")
 //   -> docker run busybox ping www.google.com
 func (self *realFramework) Run(args DockerRunArgs, cmd ...string) string {
+	dockerCommand := append(append(append([]string{"docker", "run", "-d"}, args.Args...), args.Image), cmd...)
+
+	var output string
 	if self.host.Host == "localhost" {
 		// Just run locally.
-		out, err := exec.Command("docker", append(append(append([]string{"run", "-d"}, args.Args...), args.Image), cmd...)...).CombinedOutput()
+		out, err := exec.Command("sudo", dockerCommand...).Output()
 		if err != nil {
 			self.t.Fatalf("Failed to run docker container with run args %+v due to error: %v and output: %q", args, err, out)
 			return ""
 		}
-		// The last lime is the container ID.
-		elements := strings.Split(string(out), "\n")
-		if len(elements) < 2 {
-			self.t.Fatalf("Failed to find Docker container ID in output %q", out)
+		output = string(out)
+	} else {
+		// We must SSH to the remote machine and run the command.
+		out, err := exec.Command("gcutil", append([]string{"ssh", self.host.GceInstanceName, "sudo"}, dockerCommand...)...).Output()
```

Comment:
Cant you provide a '-h' option to the docker client to make it spawn a container on a remote docker server?